staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -412,11 +420,15 @@ func (s *GenericAPIServer) installAPIResources(apiPrefix string, apiGroupInfo *A
 		apiGroupVersion.OpenAPIModels = openAPIModels
 		apiGroupVersion.MaxRequestBodyBytes = s.maxRequestBodyBytes
 
-		if err := apiGroupVersion.InstallREST(s.Handler.GoRestfulContainer); err != nil {
+		r, err := apiGroupVersion.InstallREST(s.Handler.GoRestfulContainer)
+		if err != nil {
 			return fmt.Errorf("unable to setup API %v: %v", apiGroupInfo, err)
 		}
+		resources = append(resources, r...)
 	}
 
+	s.StorageVersion.AddResourceInfo(resources...)
```

Comment:
Does this happen before we start listening?  if so, add a comment that this approach is safe because of this.